### ISSUE TITLE
Request: Use the Alpine version of BusyBox for better vulnerability remediation

### ISSUE THREAD

**Issue by liam-verta** [prometheus/busybox#50]
The Alpine project is very responsive to vulnerability reports and has been releasing patched versions of BusyBox that address critical vulnerabilities.  CVE-2022-28391 was reported over 4 months ago and is still unpatched in BusyBox 1.34.x and 1.35.0 releases.  Meanwhile, Alpine patched their BusyBox build almost as soon as the vulnerability was published: https://gitlab.alpinelinux.org/alpine/aports/-/issues/13661

This image doesn't have to use Alpine in general but it should consider using the Alpine build of the BusyBox executable for greater security.

**Comment by SuperQ** [comment]
Hmm, I'm not totally opposed to this. But the question is, how to extract/build this.

**Comment by discordianfish** [comment]
We might also finally just switch to a from scratch image.. The node-exporter is really the last container I ever feel the need to exec into and now with ephemeral containers in kubernetes even in these situations it shouldn't matter. But it would be a breaking change..

**Comment by SuperQ** [comment]
After thinking about this some more I just don't think this is a useful change.

Basically we don't care about busybox vulnerabilities in the Prometheus ecosystem. Prometheus uses statically linked Go binaries and makes no calls that require busybox. It's basically a convenience tool inside the container to make it simple to get a shell and do basic stuff.

Any vulnerability would require already being inside the container, which is game over on its own.

**Comment by juliusv** [comment]
@SuperQ I guess one benefit could be a quality-of-life improvement for the Prometheus team where we might get fewer false positive security scanner vulnerability reports despite our docs (https://prometheus.io/docs/operating/security/#automated-security-scanners) saying that we don't want those. Other than that, no big opinion either way.

**Comment by juliusv** [comment]
And then I think going completely distroless (like https://github.com/GoogleContainerTools/distroless) was being discussed, but some people still wanted a shell in the container image for convenience (which I get)?

**Comment by SuperQ** [comment]
Yes, I would very much prefer a distroless option as part of the Prometheus pipelines.

**Comment by SuperQ** [comment]
Thanks for the idea. But I think the consensus is that we want to go distroless as the best option for higher container security.

**Comment by SuperQ** [comment]
FYI, distroless is now a thing: https://github.com/prometheus/prometheus/pull/17876
